contents/blog/clickhouse-materialized-columns.md

```diff
@@ -0,0 +1,126 @@
+---
+date: 2021-09-09
+title: Speeding up ClickHouse with Materialized Columns
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+categories: engineering
+author: karl-aksel-puulmann
+featuredImage: ../images/blog/automating-software-company-github-actions.png
+featuredImageType: full
+---
+
+One quite obscure feature of ClickHouse is Materialized Columns, which allows dynamically creating columns using an expression. In this post, I’ll walk through an example query optimization in which materialized columns work well.
```

Comment:
Start with a compelling hook

> which allows dynamically creating columns using an expression

Why do I care about this?

I'd begin this post with the problem we were facing at PostHog, and then how materialized columns solved it (and the specific speedup numbers).

Sample idea:

2500%. That's one of the best case performance upgrades we got after implementing Materialized Columns in Clickhouse. There's atleast one user of PostHog who's crying with joy, since their dashboards load in a second, instead of the 25 seconds earlier.

Materialized Columns are a quite obscure feature of ClickHouse. In this post, I'll walk through an example query which is very similar to our production queries, and show you how this optimization works.

Reply:
Took inspiration from this, but my tonal choice is more toned down. Let me know what you think!